sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs

```diff
@@ -38,16 +40,24 @@ public TelemetryItem(Activity activity, ref ActivityTagsProcessor activityTagsPr
                     Tags[ContextTagKeys.AiOperationName.ToString()] = activity.DisplayName.Truncate(SchemaConstants.Tags_AiOperationName_MaxLength);
                 }
 
-                // Set ip in case of server spans only.
                 if (activity.Kind == ActivityKind.Server)
                 {
-                    var locationIp = AzMonList.GetTagValue(ref activityTagsProcessor.MappedTags, SemanticConventions.AttributeClientAddress)?.ToString();
+                    var locationIp = microsoftClientIp ??
+                                     AzMonList.GetTagValue(ref activityTagsProcessor.MappedTags, SemanticConventions.AttributeClientAddress)?.ToString();
+
                     if (locationIp != null)
                     {
                         Tags[ContextTagKeys.AiLocationIp.ToString()] = locationIp;
                     }
                 }
             }
+            else // dependency
+            {
+                if (microsoftClientIp != null) // not likely a customer would set this, but just in case
```

Comment:
nit: Remove this comment "// not likely a customer would set this, but just in case" and "// in case a customer manually puts this on a dependency or request" in activitytagsprocessor.